src/components/headers/collection-header.tsx

```diff
@@ -639,6 +662,73 @@ export class CollectionHeader extends React.Component<IProps, IState> {
     return versions.slice(0, count);
   }
 
+  private async submitCertificate(file: File) {
+    const version = this.state.versionToUploadCertificate;
+    const response = await Repositories.getRepository({
+      name: this.context.selectedRepo,
+    });
+    const signed_collection = `${PULP_API_BASE_PATH}content/ansible/collection_versions/${version.id}/`;
+    console.log(
+      'signature upload started for',
+      version,
+      response.data.results[0].pulp_href,
+      signed_collection,
+    );
+
+    this.setState({
+      alerts: this.state.alerts.concat({
+        id: 'upload-certificate',
+        variant: 'info',
+        title: t`The certificate for "${version.namespace} ${version.name} v${version.version}" is being uploaded.`,
+      }),
+    });
+
+    this.closeUploadCertificateModal();
+
+    CertificateUploadAPI.upload({
+      file,
+      repository: response.data.results[0].pulp_href,
+      signed_collection,
+    })
+      .then((result) => {
+        // This is a hack because it returns the full task api path:
+        // eg.: /api/automation-hub/pulp/api/v3/tasks/0be64cb4-3b7e-4a6b-b35d-c3b589923a90/
+        return waitForTask(result.data.task.slice(0, -1).split('/').pop()).then(
```

Comment:
```suggestion
        return waitForTask(parsePulpIDFromURL(result.data.task)).then(
```

(+ `import { parsePulpIDFromURL } from 'src/utilities';`)